File: src/main/java/Piruru/cards/AttackAndMill.java
package Piruru.cards;

import Piruru.actions.MillAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import static Piruru.Piruru.makeID;

public class AttackAndMill extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(AttackAndMill.class));
    private static final int COST = 1;
    private static final CardType TYPE = CardType.ATTACK;
    private static final CardRarity RARITY = CardRarity.COMMON;
    private static final CardTarget TARGET = CardTarget.ENEMY;
    private static final int DAMAGE = 9;
    private static final int DAMAGE_UP = 3;
    private static final int MILL_AMT = 4;
    private static final int MILL_UP = 4;

    public AttackAndMill() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, DAMAGE_UP, 0, MILL_UP, COST);
        baseDamage = damage = DAMAGE;
        baseMagicNumber = magicNumber = MILL_AMT;
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        damage(m);
        act(new MillAction(magicNumber));
    }
}

File: src/main/java/Piruru/cards/AttackAndMill.kt
package Piruru.cards

import Piruru.Piruluk.Statics.makeID
import Piruru.abstracts.PiruruCard
import Piruru.actions.MillAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.localization.CardStrings
import com.megacrit.cardcrawl.monsters.AbstractMonster


class AttackAndMill : PiruruCard(cardStrings, COST, TYPE, RARITY, TARGET, DAMAGE_UP, 0, MILL_UP, COST) {
    init {
        damage = DAMAGE
        baseDamage = damage
        magicNumber = MILL_AMT
        baseMagicNumber = magicNumber
    }

    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        damage(m)
        act(MillAction(magicNumber))
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(AttackAndMill::class.java))
        private val COST = 1
        private val TYPE = AbstractCard.CardType.ATTACK
        private val RARITY = AbstractCard.CardRarity.COMMON
        private val TARGET = AbstractCard.CardTarget.ENEMY
        private val DAMAGE = 9
        private val DAMAGE_UP = 3
        private val MILL_AMT = 4
        private val MILL_UP = 4
    }
}
